tools/python/convert_onnx_models_to_ort.py

```diff
@@ -142,18 +147,20 @@ def parse_args():
 def main():
     args = parse_args()
 
-    model_path = args.model_path.resolve()
+    model_path_or_dir = args.model_path_or_dir.resolve()
     custom_op_library = args.custom_op_library.resolve() if args.custom_op_library else None
 
-    if not model_path.is_dir():
-        raise FileNotFoundError('Model path {} is not a directory.'.format(model_path))
+    if not model_path_or_dir.is_dir() and not model_path_or_dir.is_file():
+        raise FileNotFoundError("Model path '{}' is not a file or directory.".format(model_path_or_dir))
 
     if custom_op_library and not custom_op_library.is_file():
         raise FileNotFoundError("Unable to find custom operator library '{}'".format(custom_op_library))
 
     optimization_level = _get_optimization_level(args.optimization_level)
-    _convert(model_path, optimization_level, args.use_nnapi, custom_op_library, args.save_optimized_onnx_model)
-    _create_config_file_from_ort_models(model_path, args.enable_type_reduction)
+    _convert(model_path_or_dir, optimization_level, args.use_nnapi, custom_op_library, args.save_optimized_onnx_model)
+    ort_model_path_or_dir = \
+        model_path_or_dir if model_path_or_dir.is_dir() else _onnx_model_path_to_ort_model_path(model_path_or_dir)
+    _create_config_file_from_ort_models(ort_model_path_or_dir, args.enable_type_reduction)
 
```

Comment:
Not sure using the current default config file name works that well when converting a single file (unless it's the only .onnx file in that directory - at which point you don't need to be able to convert a single onnx file) as the relationship between the config file and the source model isn't that explicit. This was one of the reasons I didn't complete adding support for providing a single model filename as input. 

If we want to support converting a single file should we require a config filename to be provided? Or maybe include the source model name in the default config filename?

Reply:
including the single model name in the config file name sounds good

---
In reply to: [595747429](https://github.com/microsoft/onnxruntime/pull/7019#discussion_r595747429) [](ancestors = 595747429)